.travis.yml

```diff
@@ -101,210 +76,197 @@ _helpers:
+  stage: &stage_test_osx_name test under OS X (last chance to fail before deploy available)
+  os: osx
+  language: shell
   env:
-  - &env_os1011_msg Build and deploy to PYPI of OS X 10.11 binary wheel=
-  - &env_py36 PYTHON_VERSION=3.6.3
-  - *env_pyenv
-  - *env_path
+  - &env_pyenv PYENV_ROOT="$HOME/.pyenv"
+  - &env_path PATH="$PYENV_ROOT/bin:$PATH"
+  before_install:
+  - brew update
+  - |
+    if brew ls --versions pyenv > /dev/null; then
+      brew outdated pyenv || brew upgrade pyenv
+      if brew ls --versions pyenv-virtualenv > /dev/null; then
+        brew outdated pyenv-virtualenv || brew upgrade pyenv-virtualenv
+      else
+        brew install pyenv-virtualenv
+      fi
+    else
+      brew install pyenv pyenv-virtualenv
+    fi
+  - |
+    eval "$(pyenv init -)"
+    eval "$(pyenv virtualenv-init -)"
+    PYENV_PYTHON_VERSION="$(pyenv install --list \
+      | grep -E "^\\s*$PYTHON_VERSION" \
+      | sort -n -t. -k3 \
+      | tail -n1 \
+      | tr -d '[:space:]' \
+      )"
+    echo $PYENV_PYTHON_VERSION
+  - pyenv install --skip-existing --keep --verbose "$PYENV_PYTHON_VERSION"
+  - pyenv shell "$PYENV_PYTHON_VERSION"
+  - python --version
+  script: skip
+  before_cache:
+  - brew --cache
   deploy:
-    <<: *deploy_step
+    <<: *_deploy_defaults
     skip_cleanup: false
     distributions: bdist_wheel
+
+# OS X test and deploy matrix helpers
+# These are used to build a matrix, with
+# N OSX image options and M python options,
+# of the form:
+#
+#   - <<: *osx_pypi_deploy_base_<osx version>
+#     name: <build name (optional)>
+#     env: *osx_env_<python version>
+
+# OS X image options
+- &osx_pypi_deploy_base_1010
+  <<: *_osx_pypi_deploy_base
+  osx_image: xcode6.4
+- &osx_pypi_deploy_base_1011
+  <<: *_osx_pypi_deploy_base
+  osx_image: xcode7.3
 - &osx_pypi_deploy_base_1012
-  <<: *osx_pypi_deploy_base_1011
-  osx_image: xcode8.1
-  env:
-  - &env_os1012_msg Build and deploy to PYPI of OS X 10.12 binary wheel=
-  - *env_py36
+  <<: *_osx_pypi_deploy_base
+  osx_image: xcode8.3
+
+# OS X Python options (the _osx_env_<version>
+# key is only needed to provide a container for
+# the &osx_env_<version> YAML anchor).
+- &osx_env_py35
+  - PYTHON_VERSION: "3.5"
   - *env_pyenv
   - *env_path
-- &osx_pypi_deploy_base_1010
-  <<: *osx_pypi_deploy_base_1011
-  osx_image: xcode6.4
-  env:
-  - &env_os1010_msg Build and deploy to PYPI of OS X 10.10 binary wheel=
-  - *env_py36
+- &osx_env_py36
+  - PYTHON_VERSION: "3.6"
+  - *env_pyenv
+  - *env_path
+- &osx_env_py37
+  - PYTHON_VERSION: "3.7"
   - *env_pyenv
   - *env_path
 
+- &linux_pypi_deploy_base
+  <<: *_generic_deploy_base
+  <<: *mainstream_python_base
+  stage: &deploy_stage_name Linux build and deploy
+  deploy:
+    <<: *_deploy_defaults
+    # `skip_cleanup: true` is required to preserve binary wheels, built
+    # inside of manylinux1 docker container during `script` step above.
+    skip_cleanup: true
+
+
 os: linux
 
-jobs:
+matrix:
```

Comment:
I prefer `jobs` here. It's intentional.

Reply:
Perhaps, but Travis logs it as:

>  `root: jobs` is an alias for `matrix` (key), using `matrix` 

I was trying to remove all travis config warnings and info lines, so that any future issues stand out immediately.

Reply:
AFAIK warnings are only shown if there's `version: "= 0"` present. Since you removed that line, Travis won't show any warnings.